tracing/zipkin/http_test.go

```diff
@@ -0,0 +1,160 @@
+package zipkin_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	zipkinkit "github.com/go-kit/kit/tracing/zipkin"
+	kithttp "github.com/go-kit/kit/transport/http"
+	zipkin "github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter/recorder"
+)
+
+const (
+	testName     = "test"
+	testBody     = "test_body"
+	testTagKey   = "test_key"
+	testTagValue = "test_value"
+)
+
+func TestHttpClientTracePropagatesParentSpan(t *testing.T) {
+	rec := recorder.NewReporter()
+	defer rec.Close()
+
+	tr, _ := zipkin.NewTracer(rec)
+
+	rURL, _ := url.Parse("http://test.com")
+
+	c := kithttp.NewClient(
+		"GET",
+		rURL,
+		func(ctx context.Context, r *http.Request, i interface{}) error {
+			return nil
+		},
+		func(ctx context.Context, r *http.Response) (response interface{}, err error) {
+			return nil, nil
+		},
+	)
+
+	clientOption := zipkinkit.HTTPClientTrace(tr)
```

Comment:
Because the call to `NewClient(...).Endpoint()` yields a Go kit Endpoint we can't do this anymore...

So instead create the zipkin tracer before creating the client and just feed it as the functional option it is:
```go
clientTracer := zipkinkit.HTTPClientTrace(tr)
ep := kithttp.NewClient(method, url, encoder, decoder, clientTracer).Endpoint()
```